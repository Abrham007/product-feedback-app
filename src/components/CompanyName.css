.CompanyName {
  width: 255px;
  height: 137px;
  flex-shrink: 0;
  display: flex;
  justify-content: space-between;
  border-radius: 10px;
  background-image: url(../assets/suggestions/desktop/background-header.png);
  padding: 62px 0px 0px 24px;
}

.CompanyName__header {
  color: #fff;
  font-size: 1.25rem;
  font-weight: 700;
  letter-spacing: -0.01563rem;
}

.CompanyName__paragraph {
  color: #fff;
  font-size: 0.9375rem;
  font-weight: 500;
  opacity: 0.75;
}

.CompanyName__btn {
  background-color: transparent;
  border: none;
  outline: none;
  display: none;
}

@media screen and (max-width: 1120px) {
  .CompanyName {
    width: 223px;
    height: 178px;
    background-image: url(../assets/suggestions/tablet/background-header.png);
    padding-top: 103px;
  }
}

@media screen and (max-width: 768px) {
  .CompanyName {
    position: fixed;
    top: 0;
    left: 0;
    width: 100vw;
    height: 72px;
    background-image: url(../assets/suggestions/mobile/background-header.png);
    background-size: cover;
    padding: 16px 24px;
    border-radius: 0px;
    visibility: visible;
  }

  .CompanyName__header {
    font-size: 0.9375rem;
    letter-spacing: -0.01169rem;
  }

  .CompanyName__paragraph {
    font-size: 0.8125rem;
  }

  .CompanyName__btn {
    display: block;
  }
}
