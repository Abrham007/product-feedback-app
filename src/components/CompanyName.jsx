import hamburgerIcon from "../assets/shared/mobile/icon-hamburger.svg";
import closeIcon from "../assets/shared/mobile/icon-close.svg";
import "./CompanyName.css";

function CompanyName(props) {
  //  This function handles the click of the hamberger menu
  //  and shows the suggestions board in the suggestions component
  function handleClick() {
    props.setIsOpen((prevValue) => !prevValue);
  }
  return (
    <div className="CompanyName">
      <div>
        <h1 className="CompanyName__header">Your Company</h1>
        <p className="CompanyName__paragraph">Feedback Board</p>
      </div>
      <button onClick={handleClick} className="CompanyName__btn">
        <img
          src={props.isOpen ? closeIcon : hamburgerIcon}
          alt="menu icon"
          width={20}
          height={17}
        ></img>
      </button>
    </div>
  );
}

export default CompanyName;
