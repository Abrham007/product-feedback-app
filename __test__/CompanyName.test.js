import React from "react";
import renderer from "react-test-renderer";
import CompanyName from "../src/components/CompanyName";

describe("CompanyName", () => {
  it("renders the correct JSX components", () => {
    const component = renderer.create(
      <CompanyName isOpen={false} setIsOpen={(prevValue) => !prevValue} />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
