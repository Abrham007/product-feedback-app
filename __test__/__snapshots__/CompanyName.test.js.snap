// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CompanyName renders the correct JSX components 1`] = `
<div
  className="CompanyName"
>
  <div>
    <h1
      className="CompanyName__header"
    >
      Your Company
    </h1>
    <p
      className="CompanyName__paragraph"
    >
      Feedback Board
    </p>
  </div>
  <button
    className="CompanyName__btn"
    onClick={[Function]}
  >
    <img
      alt="menu icon"
      height={17}
      width={20}
    />
  </button>
</div>
`;
